src/Contract/StartBidding.purs

```diff
@@ -215,28 +216,18 @@ mkStartBiddingContractWithErrors (StartBiddingContractParams params) = do
     , constraints = constraints
     }
 
-queryAuctionEscrowUtxo :: AuctionInfo -> Contract (Maybe Utxo)
-queryAuctionEscrowUtxo (AuctionInfo auctionInfo) =
-  utxosAt auctionInfo.auctionEscrowAddr
-    <#> Array.find (isCurrentAuctionEscrowUtxo <<< _.output <<< unwrap <<< snd)
-    <<< Map.toUnfoldable
+queryAuctionEscrowUtxo :: AuctionEscrowState -> AuctionInfo -> Contract (Maybe Utxo)
+queryAuctionEscrowUtxo escrowState (AuctionInfo auctionInfo) =
+  utxosAt auctionInfo.auctionEscrowAddr <#>
+    (Array.find (worker <<< _.output <<< unwrap <<< snd) <<< Map.toUnfoldable)
   where
-  AuctionTerms auctionTerms = auctionInfo.auctionTerms
-
-  isCurrentAuctionEscrowUtxo :: TransactionOutput -> Boolean
-  isCurrentAuctionEscrowUtxo (TransactionOutput txOut) =
-    auctionEscrowValue `Value.leq` txOut.amount
-      && (txOut.datum == OutputDatum (wrap $ toData AuctionAnnounced))
-      && isNothing txOut.referenceScript
-
-  auctionEscrowValue :: Value
-  auctionEscrowValue =
-    auctionTerms.auctionLot <> mkAuctionToken auctionEscrowTokenName
-      <> mkAuctionToken standingBidTokenName
-
-  mkAuctionToken :: TokenName -> Value
-  mkAuctionToken tokenName =
-    Value.singleton auctionInfo.auctionId tokenName one
+  auctionCs :: CurrencySymbol
+  auctionCs = auctionInfo.auctionId
+
+  worker :: TransactionOutput -> Boolean
+  worker (TransactionOutput txOut) =
```

Comment:
I guess the name `worker` here might have been more self-descriptive.

Reply:
Agreed, renamed to `isValidAuctionEscrowUtxo`.